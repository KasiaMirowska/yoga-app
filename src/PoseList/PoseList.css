
.pose-list {
    background-color: rgb(233, 224, 231); 
    background-color: rgb(220, 235, 228); 
    
    text-align: center; 
    padding-top: 30px; 
    padding-bottom: 80px;
}
.title {
    font-size: 1.8em;
    display: block;
    padding: 5px;
}

@media (min-width: 600px) {
    .poses-container {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-around; 
        width: 60%;
        margin: auto;
    }
  }  