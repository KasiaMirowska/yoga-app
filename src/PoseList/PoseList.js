import React from 'react';
import YogaContext from '../Context';
import PoseListItem from '../PoseListItem/PoseListItem';
import APIPoseCalls from '../services/API_Pose_service';
import './PoseList.css';

export default class PoseList extends React.Component {
    static contextType = YogaContext;
    
    
    componentDidMount = () => {
        APIPoseCalls.getAllPosesData()
            .then(data => {
                this.context.setPosesList(data);
            })
            .catch(err => {
                this.context.setError(err)
            })
    }

    render() {
        const poses = this.context.poses.map(pose => {
            return <PoseListItem
                key={pose.id}
                id={pose.id}
                name={pose.name_eng}
                sanskrit={pose.name_san}
                img={pose.img}
            />

        })
        return (
            <div className='pose-list'>
                <h2 className='title' >ALL POSES: </h2>
                <ul className='poses-container'>
                {poses}
                </ul>
               
            </div>
        )
    }
}