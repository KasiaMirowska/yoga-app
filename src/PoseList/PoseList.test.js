import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import PoseList from './PoseList';
import {BrowserRouter as BR } from 'react-router-dom';


describe('PoseList component', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<BR><PoseList /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const list = renderer.create(<BR><PoseList /></BR>);
        expect(list.toJSON()).toMatchSnapshot();
    })
})