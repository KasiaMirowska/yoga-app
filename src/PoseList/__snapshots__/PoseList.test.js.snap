// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoseList component renders UI as expected 1`] = `
<div
  className="pose-list"
>
  <div
    className="error"
  />
  <h2
    className="title"
  >
    POSES LIBRARY: 
  </h2>
  <ul
    className="poses-container"
  />
</div>
`;
